gnucash/import-export/ofx/gnc-ofx-import.c

```diff
@@ -1091,22 +1094,22 @@ void gnc_file_ofx_import (GtkWindow *parent)
         selected_filename = conv_name;
 #endif
 
-        DEBUG("Opening selected file");
+        DEBUG("Opening selected file(s)");
         num_trans_processed = 0;
-        libofx_proc_file(libofx_context, selected_filename, AUTODETECT);
-        // Now would be a good time to see whether the view has anything in it!
+        for (iter = selected_filenames; iter; iter = iter->next)
+            libofx_proc_file (libofx_context, iter->data, AUTODETECT);
         if(gnc_gen_trans_list_empty(gnc_ofx_importer_gui))
         {
             gnc_gen_trans_list_delete (gnc_ofx_importer_gui);
             if(num_trans_processed)
-                gnc_info_dialog(parent,_("OFX file imported, %d transactions processed, no transactions to match"),num_trans_processed);
+                gnc_info_dialog (parent, _("OFX file(s) imported, %d transactions processed, no transactions to match"), num_trans_processed);
         }
         else
         {
             gnc_gen_trans_list_show_all(gnc_ofx_importer_gui);
         }
-        g_free(selected_filename);
```

Comment:
Why remove the g_free here, maybe I am misunderstanding this.

Reply:
My bad, g_slist_free() by default does not free the strings themselves.

Reply:
This construct is slightly complex (no blame, it was already before you started working on it). I would like this better if the GSList functions would be used to fully manage the list. I mean if list memory is freed with `g_slist_free_full`. Also do this as soon as you no longer need the memory, which is right after the for loop.

The non-Windows case is straightforward. It can work directly with `iter->data` (via `selected_filename` as is set right before the Windows block. With a `g_slist_free_full` right after the for loop, memory is handled cleanly.

The Windows case needs special treatment because we need to localize the name. Which means we create a variable that needs to be freed. In the current Windows block, just reset `selected_filename` to the result of `g_win32_locale_filename_from_utf8`. No need for a temporary variable and don't free anything there. Instead at the very end of the for loop, add another Windows only block and `g_free` the `selected_filename` variable in there.

That makes the flow on each platform more consistent:
* non-windows: the only memory management is done by `g_slist_free_full`.
* on windows: the list is memory-managed by the same `g_slist_free_full` and the localized variable by a `g_free`.